src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useRef, useEffect } from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  deleteTodo: (id: number) => void;
+  handleToggle: (
+    event: React.ChangeEvent<HTMLInputElement>, id: number) => void;
+  editingTitle: (id: number, title: string) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  handleToggle,
+  editingTitle,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [editingValue, setEditingValue] = useState(todo.title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  return (
+    <li
+      className={classNames({
+        completed: todo.completed,
+        editing: isEditing,
+      })}
+      hidden={false}
+      onDoubleClick={(event) => {
+        if (event.target !== inputRef.current) {
+          setIsEditing(true);
+        }
+      }}
+    >
+      { isEditing ? (
+        <input
+          type="text"
+          ref={inputRef}
+          className="edit"
+          value={editingValue}
+          onChange={(event) => {
+            setEditingValue(event.target.value);
+          }}
+          onKeyDown={(event) => {
+            if (event.key === 'Enter') {
+              if (editingValue.trim() === '') {
+                deleteTodo(todo.id);
+              } else {
+                editingTitle(todo.id, editingValue);
+                setIsEditing(false);
+              }
+            } else if (event.key === 'Escape') {
+              setIsEditing(false);
+              setEditingValue(todo.title);
+            }
+          }}
```

Comment:
it would be better to create separate function and keep jsx clean
```suggestion
          onKeyDown={onKeyDown}
```